eng/OffByDefaultRoslynComponent.targets.template

```diff
@@ -0,0 +1,17 @@
+<Project>
+    <Target Name="_{TargetPrefix}GatherAnalyzers">
+        <ItemGroup>
+            <_{TargetPrefix}Analyzer Include="@(Analyzer)" Condition="'%(Analyzer.NuGetPackageId)' == '{NuGetPackageId}'" />
+        </ItemGroup>
+    </Target>
```

Comment:
https://github.com/dotnet/runtime/blob/main/eng/MultiTargetRoslynComponent.targets.template doesn't have a separate target to gather the analyzers. Avoid the separate target as it isn't necessary and adds cost to the build.

Instead do the following in L14:

```xml
<Analyzer Remove="@(Analyzer->WithMetadataValue('NuGetPackageId', '{NuGetPackageId}'))" />
```

Reply:
Oh, clever!